commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPI.java

```diff
@@ -112,6 +112,11 @@ public static void onLoad(CommandAPIConfig<?> config) {
 			CommandAPIPlatform<?, ?, ?> platform = CommandAPIVersionHandler.getPlatform();
 			new CommandAPIHandler<>(platform);
 
+			if (CommandAPI.getConfiguration().shouldUseLatestNMSVersion() || CommandAPI.getConfiguration().shouldBeLenientForMinorVersions()) {
+				CommandAPI.logWarning("Loading the CommandAPI with a potentially incompatible NMS implementation.");
+				CommandAPI.logWarning("While you may find success with this, further updates might be necessary to fully support the version you are using.");
+			}
+
```

Comment:
This may mistakenly log the warning when `shouldUseLatestNMSVersion = false`, `shouldBeMoreLenientForMinorVersions = true`, and the plugin is being loaded on a supported version. For example, if I set the config `lenient-for-minor-versions: true` and run the plugin on Paper 1.20, I get this log:

```
[10:04:46 INFO]: [CommandAPI] Loading server plugin CommandAPI v9.6.0-SNAPSHOT
[10:04:46 WARN]: [CommandAPI] Loading the CommandAPI with a potentially incompatible NMS implementation.
[10:04:46 WARN]: [CommandAPI] While you may find success with this, further updates might be necessary to fully support the version you are using.
[10:04:46 INFO]: [CommandAPI] Loaded platform NMS_1_20_R1 > NMS_CommonWithFunctions > NMS_Common > CommandAPIBukkit
```

This seems misleading, as `NMS_1_20_R1` _is_ compatible with Paper 1.20 (it is not potentially incompatible).

Additionally, I'm not sure if it makes sense to put this warning in the `CommandAPI` class, since this code will also run on Velocity. Velocity does not do anything if these options are enabled (for example when shading, since `CommandAPIConfig` provides access to these methods on Velocity), so it will always load a compatible platform.

Perhaps these warning messages could be placed in `commandapi-bukit-vh`

```java
if (shouldUseLatestNMSVersion) {
    warn("Forcefully loading the latest NMS version. This is not guaranteed to work.")
    return NMS_1_21();
}

return switch(version) {
    case "1.16.5" -> NMS_1_16_5();
    // ...
    default -> {
        if (shouldBeMoreLeinientForMinorVersions()) {
            warn("Attempting to load a minor NMS version. This is not guaranteed to work.")
            return NMS_1_21();
        } else {
            throw UnsupportedVersion();
        }
    }
}
```

Reply:
Well, I originally put this in the version handler but apparently the logger needs a platform to work so I put it after the initialisation of the CommandAPIHandler instance.

What I could do is adjust the return type for the getPlatform method a bit to hold more context, such as the latest supported version and of course the CommandAPIPlatform instance?

Reply:
Ohhhh, yea, hmmm, that's tricky...

Maybe the extra context could be a `Consumer` to call back like so?

```java
record LoadInfo(CommandAPIPlatform<?, ?, ?> platform, Runnable postLoadCallback);

static LoadInfo getPlatform() {
    if (shouldUseLatestNMSVersion) {
        return new LoadInfo(new NMS_1_21_R1(), () -> {
            CommandAPI.logWarning("Forcefully loading the latest NMS version. This is not guaranteed to work.");
        });
    }

    CommandAPIPlatform<?, ?, ?> platform = switch (version) {
        case "1.16.5" -> new NMS_1_16_R3();
        // ...
        case "1.21", "1.21.1" -> new NMS_1_21_R1();
        default -> null;
    };

    if (platform != null) {
        // Normal load, nothing to do
        return new LoadInfo(platform, () -> {});
    }

    if (shouldBeLenientForMinorVersions) {
        return new LoadInfo(new NMS_1_21_R1(), () -> {
            CommandAPI.logWarning("Attempting to load a minor NMS version. This is not guaranteed to work.");
        });
    } else {
        throw new UnsupportedVersionException(version);
    }
}
```

Reply:
Yeah, that's a good idea.